test/functional/forall/reduce-basic/CMakeLists.txt

```diff
@@ -0,0 +1,29 @@
+###############################################################################
+# Copyright (c) 2016-20, Lawrence Livermore National Security, LLC
+# and RAJA project contributors. See the RAJA/COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (BSD-3-Clause)
+###############################################################################
+
+#
+# List of reduction types for generating test files.
+#
+set(REDUCETYPES ReduceSum ReduceMin ReduceMax ReduceMinLoc ReduceMaxLoc)
+
+
+#
+# Generate tests for each enabled RAJA back-end
+#
+# Note: FORALL_BACKENDS is defined in ../CMakeLists.txt
+#
+foreach( BACKEND ${FORALL_BACKENDS} )
+  foreach( REDUCETYPE ${REDUCETYPES} )
+    configure_file( test-forall-basic-reduce.cpp.in
+                    test-forall-basic-${REDUCETYPE}-${BACKEND}.cpp )
+    raja_add_test( NAME test-forall-basic-${REDUCETYPE}-${BACKEND}
+                   SOURCES test-forall-basic-${REDUCETYPE}-${BACKEND}.cpp )
```

Comment:
What happens when hip looks for test-forall-basic-${REDUCETYPE}-${BACKEND}.cpp is that it looks in ${CMAKE_CURRENT_SOURCE_DIR} when it should be looking in ${CMAKE_CURRENT_BINARY_DIR}.
So this should work if we add that to the source.
```
    raja_add_test( NAME test-forall-basic-${REDUCETYPE}-${BACKEND}
                   SOURCES ${CMAKE_CURRENT_BINARY_DIR}test-forall-basic-${REDUCETYPE}-${BACKEND}.cpp )
```

Reply:
@davidbeckingsale mentioned that may work, but I forgot to try it. Do you need a forward slash before 'test-forall...'?

Reply:
Yes, I'll update the snippet.
That is working for me on ubuntu with rocm 3.5

Reply:
So I added the binary dir path to the test target and this fixes the hip builds.